packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts

```diff
@@ -159,4 +161,35 @@ export class StateGraph {
     }
   }
 
+  /**
+   * Binds this StateGraph to the StateMachine it defines and updates state machine permissions
+   */
+  public bind(stateMachine: StateMachine) {
```

Comment:
Nice! <3

Reply:
I was able to figure out how to use a separate inline policy to lock this down to state machine arn.  The existing `addRoleToPolicy` or `grantStartExecution` methods use the default policy for the role which results in a circular reference, but a separate inline policy does not have the same circular reference issue.